### PR TITLE
signalbackup-tools: 20240816 -> 20240830

### PR DESCRIPTION
**Pull request by malob**
## Description of changes

Upadate to latest version. Add new Darwin specific framework dependency and update to using `cmake`.

Diff: https://github.com/bepaald/signalbackup-tools/compare/20240816...20240830

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by NickCao — COMMENTED** [review]
Shall we switch to cmake?

**Comment by malob** [comment]
@NickCao, didn't notice he added that. Good call. Done.

**Review by NickCao — COMMENTED** [review]
Can we drop `postPatch` and `installPhase`?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -12502,7 +12502,9 @@ with pkgs;
   sigil = libsForQt5.callPackage ../applications/editors/sigil { };
 
   signalbackup-tools = darwin.apple_sdk_11_0.callPackage
-    ../applications/networking/instant-messengers/signalbackup-tools { };
+    ../applications/networking/instant-messengers/signalbackup-tools {
+      inherit (darwin.apple_sdk.frameworks) Security;
```
Comment (NickCao) [inline pkgs/top-level/all-packages.nix:12506]: Please use `darwin.apple_sdk.frameworks.Security` directly, see https://github.com/NixOS/nixpkgs/issues/204303

Comment (malob) [reply pkgs/top-level/all-packages.nix]: Thanks, I wasn't aware of that. Fixed.

**Comment by malob** [comment]
@NickCao, good catch regarding `postPatch` that was a sloppy oversight on my part. As for `installPhase`, pretty sure it's needed. Package doesn't build without it. 

I'm no CMake expert, but pretty sure this package's CMake config doesn't contain any install instructions or whatever.

**Review by NickCao — APPROVED** [review]
diff LGTM, waiting for ofborg to build on darwin.

**Comment by malob** [comment]
Result of `nixpkgs-review pr 338253` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>signalbackup-tools</li>
  </ul>
</details>

**Pull request merged**